apps/website/src/pages/profile.tsx

```diff
@@ -119,6 +194,114 @@ const ProfilePage = withAuth(({ auth }) => {
   );
 });
 
+type ProfileAccountDetailsProps = {
+  userName: string | null | undefined;
+  userEmail: string;
+  tempName: string;
+  isSaving: boolean;
+  nameError: string;
+  onFocus: () => void;
+  onNameChange: (name: string) => void;
+  onSave: () => void;
+  onCancel: () => void;
+};
+
+const ProfileAccountDetails: React.FC<ProfileAccountDetailsProps> = ({
+  userName,
+  userEmail,
+  tempName,
+  isSaving,
+  nameError,
+  onFocus,
+  onNameChange,
+  onSave,
+  onCancel,
+}) => {
+  const showButtons = tempName !== (userName || '');
+  return (
+    <>
+      <H3>Account details</H3>
+      <div className="flex flex-col gap-4 container-lined bg-white p-8 h-fit">
+        <div className="profile-name">
+          <div className="flex flex-col gap-2">
+            <P><span className="font-bold">Name:</span></P>
+            <div className="flex gap-2 items-center">
+              <Input
+                inputClassName="profile-name__input"
+                labelClassName="flex-1 min-w-0"
+                value={tempName}
+                onChange={(e) => onNameChange(e.target.value)}
+                onFocus={onFocus}
+                onKeyDown={(e) => {
+                  if (e.key === 'Enter') onSave();
+                  if (e.key === 'Escape') onCancel();
+                }}
+                placeholder="Enter your name"
+              />
+              {showButtons && (
+                <div className="profile-name__buttons flex gap-2 flex-shrink-0">
+                  <CTALinkOrButton
+                    variant="primary"
+                    onClick={onSave}
+                    disabled={isSaving}
+                    className="whitespace-nowrap"
+                  >
+                    {isSaving ? 'Saving...' : 'Save'}
+                  </CTALinkOrButton>
+                  <CTALinkOrButton
+                    variant="secondary"
+                    onClick={onCancel}
+                    disabled={isSaving}
+                  >
+                    Cancel
+                  </CTALinkOrButton>
+                </div>
+              )}
+            </div>
+            {nameError && (
+              <p className="profile-name--error text-red-600 text-size-sm mt-1">{nameError}</p>
+            )}
+          </div>
+        </div>
+        <P><span className="font-bold">Email:</span> {userEmail}</P>
+      </div>
+    </>
+  );
+};
+
+type ProfileCourseListProps = {
+  enrolledCourses: {
+    course: Course;
+    courseRegistration: CourseRegistration;
+  }[];
+};
+
+const ProfileCourseList: React.FC<ProfileCourseListProps> = ({ enrolledCourses }) => {
+  return (
+    <>
+      <H3>Your courses</H3>
+      {enrolledCourses.length === 0 && (
+        <div className="profile__no-courses flex flex-col gap-4 container-lined bg-white p-8 mb-4">
```

Comment:
this block becomes "profile-course-list" as it is now in its own component outside "profile"

```suggestion
        <div className="profile-course-list__no-courses flex flex-col gap-4 container-lined bg-white p-8 mb-4">
```
```suggestion
        <div className="profile__no-courses flex flex-col gap-4 container-lined bg-white p-8 mb-4">
```

Reply:
unresolving; i dont see this updated? it's just a small nit but LMK if you disagree here